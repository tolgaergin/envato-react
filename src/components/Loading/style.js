import styled, { keyframes } from 'styled-components';

export const Container = styled.div `
  display: flex;
  align-items: center;
  justify-content: center;
  height: 100%;
`;

export const Dots = styled.div `
  display: flex;
  justify-content: center;
  font-size: 22px;
`;

const loadingFade = keyframes `
  0% { opacity: 0; }
  50% { opacity: 0.8; }
  100% { opacity: 0; }
`;

export const Dot = styled.div `
  width: 8px;
  height: 8px;
  margin: 0 4px;
  background-color: #000;
  border-radius: 50%;
  opacity: 0;
  box-shadow: 0 0 2px #fff;
  animation: ${loadingFade} 1s infinite;
  animation-delay: 0s;

  &:nth-child(2) {
    animation-delay: 0.1s;
  }

  &:nth-child(3) {
    animation-delay: 0.2s;
  }
`;
