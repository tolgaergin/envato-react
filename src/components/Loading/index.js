import React from 'react';

import { Container, Dots, Dot } from './style';

const Loading = () => (
  <div className="child">
    <Container>
      Loading
      <Dots>
        <Dot />
        <Dot />
        <Dot />
      </Dots>
    </Container>
  </div>
);

export default Loading;
