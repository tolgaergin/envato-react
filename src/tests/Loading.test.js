import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Loading from '../components/Loading/index';

describe('Loading', () => {
  it('renders and matches snapshot', () => {
    const component = renderer.create(
      <Loading />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
