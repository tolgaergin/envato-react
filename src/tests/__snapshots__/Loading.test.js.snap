exports[`Loading renders and matches snapshot 1`] = `
<div
  className="child">
  <div
    className="ivhEYB">
    Loading
    <div
      className="cYmFlc">
      <div
        className="cbekqL" />
      <div
        className="cbekqL" />
      <div
        className="cbekqL" />
    </div>
  </div>
</div>
`;
